Sources/Logging/Strings/PurchaseStrings.swift

```diff
@@ -208,11 +208,12 @@ extension PurchaseStrings: LogMessage {
             return "There is a problem with the SKPayment missing " +
             "a product identifier - this is an issue with the App Store."
 
-        case .sktransaction_missing_transaction_date:
+        case let .sktransaction_missing_transaction_date(transactionState):
             return """
             The SKPaymentTransaction has a nil value for transaction date - this is a bug in StoreKit.
             Unix Epoch will be used instead for the transaction within the app.
             Transactions in the backend and in webhooks are unaffected and will have the correct timestamps.
+            Transaction state: \(transactionState)
```

Comment:
I added this for awareness, but we'll eventually need to deal with this and remove this log.
[The docs](https://developer.apple.com/documentation/storekit/skpaymenttransaction/1411273-transactiondate) make this clear:
> The contents of this property are undefined except when `transactionState` is set to `SKPaymentTransactionState.purchased` or `SKPaymentTransactionState.restored`.